examples/ramen_rec/tests/test_constraint_solving.py

```diff
@@ -15,48 +18,63 @@ def test_choose_ramens_using_constraints(
         )
     )
 
-    solution = (
-        ConstraintSolver()
-        .set_candidates(candidates=pipeline_candidate_results)
-        .set_sections(sections=2)
-        .set_items_per_section(count=3)
+    solver_sections = (SectionSetConstraint(scaling=100)
```

Comment:
Same as my above comment, far too low level. The purpose of having the framework is to encapsulate all of these details. If the various generation or filtering stages need parameterization, that should be done on constructing the stage.

Reply:
test updated to just use the new pipeline